### PR TITLE
Add patches for Proxy WASM enabling (OSSM-1931)

### PR DESCRIPTION
**Pull request by dcillera**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: bazel/com_github_google_flatbuffers.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff -ruN a/include/flatbuffers/base.h b/include/flatbuffers/base.h
+--- a/include/flatbuffers/base.h	2022-09-30 18:32:00.076755650 +0200
++++ b/include/flatbuffers/base.h	2022-09-30 18:32:00.103755894 +0200
+@@ -258,6 +258,9 @@
+   #endif
+ #endif // !FLATBUFFERS_HAS_NEW_STRTOD
+ 
++// The following define added to fix OSSM-1931
++#define FLATBUFFERS_LOCALE_INDEPENDENT 0
```
Comment (jwendell) [inline bazel/com_github_google_flatbuffers.patch:9]: This can be passed as a command line argument in bazel, no?

Comment (dcillera) [reply bazel/com_github_google_flatbuffers.patch:9]: Yes I think so

**Comment by oschaaf** [comment]
CI is still busy, but in the log output I notice:

```
ERROR: /home/user/.cache/bazel/_bazel_user/004a6f22074b0cbbac00100c854c36fb/external/emsdk/BUILD:11:8: Executing genrule @emsdk//:local_link_rule failed: (Exit 127): bash failed: error executing command 
  (cd /home/user/.cache/bazel/_bazel_user/004a6f22074b0cbbac00100c854c36fb/sandbox/processwrapper-sandbox/7470/execroot/envoy && \
  exec env - \
    PATH=/bin:/usr/bin:/usr/local/bin \
  /bin/bash -c 'source external/bazel_tools/tools/genrule/genrule-setup.sh; /work/maistra/vendor/emsdk/local_link_to_host.sh; echo hello world >bazel-out/k8-opt-exec-2B5CBBC6-ST-3d434dd55ca0/bin/external/emsdk/hello.txt')
# Configuration: f99cb122082e97a2df774cbae6e6098f90d9bfdaa52902565c4aaa6aaa512de8
# Execution platform: @local_config_platform//:host

Use --sandbox_debug to see verbose messages from the sandbox
```

File: bazel/com_github_google_flatbuffers.patch (hunk 1 above)
Comment (dcillera) [reply bazel/com_github_google_flatbuffers.patch:9]: The working way I've found to do that is using the flag_set of Bazel in maistra/vendor/emsdk/emscripten_toolchain/crosstool.bzl:

flag_set(
            actions = all_compile_actions,
            flags = ['-DFLATBUFFERS_LOCALE_INDEPENDENT=0'],
        ),

Comment (jwendell) [reply bazel/com_github_google_flatbuffers.patch:9]: I was thinking something like https://github.com/maistra/envoy/blob/maistra-2.2/maistra/bazelrc#L6

Comment (dcillera) [reply bazel/com_github_google_flatbuffers.patch:9]: Yes, I had tried it, for example, in BUILD file:

time bazel_build --copt -DFLATBUFFERS_LOCALE_INDEPENDENT=0 //extensions:stats.wasm

but didn't work

Comment (dcillera) [reply bazel/com_github_google_flatbuffers.patch:9]: No success even in:
COMMON_FLAGS="\
    --copt -DFLATBUFFERS_LOCALE_INDEPENDENT=0 \
    --config=release \
    --config=${ARCH} \
"

**Comment by dcillera** [comment]
This PR will be closed as patches won't be used anymore

**Pull request closed without merging**